Cargo.toml

```diff
@@ -5,8 +5,11 @@ prettytable-rs = "^0.10"
 rand = "^0.8"
 
 [dependencies.arrow2]
+branch = "clark/expand-casting-support"
```

Comment:
The main con with working off of main is that we'd be making the divergence from upstream Arrow2 implicit, and if we keep merging upstream Arrow2 into our fork, our differing commits might be lost in the git history. Instead, I think it would be better to anchor to a branch where all of the diverging commits are at the head and try to keep the branching small/short-lived.

So the flow would be:
1. Create a branch on Arrow2 fork containing requisite changes to Arrow2.
2. Update Daft to point to that branch.
3. Submit a PR from Arrow2 fork branch to upstream Arrow2.
4. If the PR is updated during the review process, we can update the locked Daft dependency with a `cargo update` PR.
5. When the PR is merged and is included in an Arrow2 release, switch Daft's Arrow2 dependency to point to that crates.io release.
6. If we need another Arrow2 change stacked on the existing Arrow2 branch/PR, we could create a nightly snapshot branch containing both changes, similar to what Polars does: https://github.com/pola-rs/polars/blob/528590cfa57e48f5bd902ad027f5e35318644110/Cargo.toml#L47

I _think_ this should help keep the difference with upstream Arrow2 explicit, which should be nicer? Let me know what you think about that flow.